.github/workflows/build-and-release.yaml

```diff
@@ -225,6 +227,7 @@ jobs:
       - name: Publish binary
         if: startsWith(github.ref, 'refs/tags/')
         run: |
+          pip install setuptools
```

Comment:
The reason why it runs `pip install setuptools` is because it will fail to run `node-libcurl/node_modules/node-gyp/gyp/gyp_main.py` without this step.

You can get more details here [build error](https://github.com/Kong/node-libcurl/actions/runs/7161876409/job/19498083679)

Reply:
The error is `ModuleNotFoundError: No module named 'distutils'`
It would appear distutils was deprecated in python 3.10 https://peps.python.org/pep-0632/
So this error is consistent with the github action ambient version begin 3.12 or above.
Its worth mentioning that this step only fails on macos